src/app/data-planes/components/PolicyTypeEntryList.vue

```diff
@@ -0,0 +1,161 @@
+              <TagList
+                v-if="Array.isArray(rowValue) && rowValue.length > 0"
+                class="tag-list"
+                :tags="rowValue"
+              />
+
+              <template v-else>
+                —
+              </template>
+            </template>
+
+            <template #name="{ rowValue }">
+              <template v-if="rowValue !== null">
+                {{ rowValue }}
+              </template>
+
+              <template v-else>
+                —
+              </template>
+            </template>
+
+            <template #origins="{ rowValue }">
+              <div
+                v-if="rowValue.length > 0"
+                class="origin-list"
+              >
+                <router-link
+                  v-for="(origin, originIndex) in rowValue"
+                  :key="`${index}-${originIndex}`"
+                  :to="origin.route"
+                >
+                  {{ origin.name }}
+                </router-link>
+              </div>
+
+              <template v-else>
+                —
+              </template>
+            </template>
+
+            <template #config="{ rowValue, rowKey }">
+              <template v-if="rowValue !== null">
+                <CodeBlock
+                  :id="`${index}-${rowKey}-code-block`"
+                  :code="rowValue"
+                  language="yaml"
+                  :show-copy-button="false"
+                />
+              </template>
+
+              <template v-else>
+                —
+              </template>
+            </template>
+          </KTable>
+        </div>
+      </template>
+    </AccordionItem>
+  </AccordionList>
+</template>
+
+<script lang="ts" setup>
+import { PropType } from 'vue'
+import { KTable } from '@kong/kongponents'
+
+import AccordionList from '@/app/common/AccordionList.vue'
+import AccordionItem from '@/app/common/AccordionItem.vue'
+import CodeBlock from '@/app/common/CodeBlock.vue'
+import PolicyTypeTag from '@/app/common/PolicyTypeTag.vue'
+import TagList from '@/app/common/TagList.vue'
+import { PolicyTypeEntry, TableHeader } from '@/types/index'
+
+const tableHeaders: TableHeader[] = [
+  { label: 'From', key: 'sourceTags' },
+  { label: 'To', key: 'destinationTags' },
+  { label: 'On', key: 'name' },
+  { label: 'Conf', key: 'config' },
+  { label: 'Origins', key: 'origins' },
+]
+
+const props = defineProps({
+  policyTypeEntries: {
+    type: Object as PropType<PolicyTypeEntry[]>,
+    required: true,
+  },
+})
+
+function getCellAttributes({ headerKey }: any): Record<string, string> {
+  return { class: `cell-${headerKey}` }
+}
+</script>
+
+<style lang="scss" scoped>
+.policy-list {
+  display: flex;
+  flex-direction: column;
+  gap: var(--spacing-xs);
+}
+
+.tag-list {
+  display: flex;
+  margin-top: var(--spacing-xxs);
+}
+
+.origin-list {
+  display: flex;
+  flex-direction: column;
+}
+</style>
+
+<style lang="scss">
+.policy-type-table.policy-type-table td {
```

Comment:
Is this maybe your editor again? I don't think you need `policy-type-table` twice here?

Reply:
> What do you mean by this?

There's was another thing I picked up on that was just something you said your editor did (the adding `js` to the imports) which I wasn't aware of, I thought this could have been similar.

> Yes, you do. Without increasing the specificity of this rule, the KTable default overrides it to vertical-align: middle which I don’t want for this table. The cells should be top-aligned.

Oh I see 😄 , not necessary to change unless you like it better, but I wondered if something like `.policy-type-table .k-table td` might have read more like "I'm doing this specifically to overwrite a rule that is only troublesome in k-tables" - not sure if that has exactly the same effect - just the double class looked weird/jumped out to me.

Reply:
> I wondered if something like `.policy-type-table .k-table td` might have read more like "I'm doing this specifically to overwrite a rule that is only troublesome in k-tables" - not sure if that has exactly the same effect - just the double class looked weird/jumped out to me.

The effect would be the same (adding one class selector to the selector list); however, I _specifically_ duplicate an _existing_ selector if possible because in my opinion, that does the job better and with less ambiguity:

- `.policy-type-table .k-table td`: more ambiguous. It doesn’t necessarily implicate that this is a specificity hack but rather that the selector needs to be scoped to `td`s within `.k-table`s within `.policy-type-table`s. One could think "ah, but `.k-table` here isn’t needed so I can remove it". That would be wrong and it would be hard to spot.
- `.k-table.k-table td`: too general. As written, this looks to be generally applicable to KTable uses which isn’t intentional here. It so happens that this only has KTables we want to change, but that’s just circumstantial.
- `.policy-type-table.policy-type-table td`: Indicates that this is a specificity hack which as far as I know is the only reason one would want to duplicate a class selector like this. Scopes the rule to only the tables we want to scope it to.